experimental/generation/generator/packages/library/src/dialogGenerator.ts

```diff
@@ -335,23 +340,25 @@ async function processTemplate(
 
                             // See if generated file has been overridden in templates
                             let existing = await findTemplate(filename, templateDirs) as Plain
-                            if (existing?.source) {
+                            if (existing && existing.source.endsWith(ppath.normalize(filename))) {
```

Comment:
What is source is null or undefined?  Would existing?.source?.endsWith() work better here?